HARK/ConsumptionSaving/ConsRiskyAssetModel.py

```diff
@@ -239,6 +244,210 @@ def get_shocks(self):
         self.get_Adjust()
 
 
+####################################################################################################
+####################################################################################################
+
+
+@dataclass
+class ConsRiskySolver(ConsIndShockSolver):
+    solution_next: ConsumerSolution
+    IncShkDstn: DiscreteDistribution
+    TranShkDstn: DiscreteDistribution
+    PermShkDstn: DiscreteDistribution
+    RiskyDstn: DiscreteDistribution
+    LivPrb: float
+    DiscFac: float
+    CRRA: float
+    Rfree: float
+    PermGroFac: float
+    BoroCnstArt: float
+    aXtraGrid: np.array
+    vFuncBool: bool
+    CubicBool: bool
+
+    def __post_init__(self):
+        self.def_utility_funcs()
+
+        # Make sure the individual is liquidity constrained.  Allowing a consumer to
+        # borrow *and* invest in an asset with unbounded (negative) returns is a bad mix.
+        if self.BoroCnstArt != 0.0:
+            raise ValueError("RiskyAssetConsumerType must have BoroCnstArt=0.0!")
+
+        if self.CubicBool:
+            raise NotImplementedError(
+                "RiskyAssetConsumerType does not implement cubic interpolation yet!"
+            )
+
+    def set_and_update_values(self, solution_next, IncShkDstn, LivPrb, DiscFac):
+        super().set_and_update_values(solution_next, IncShkDstn, LivPrb, DiscFac)
+
+        # overwrite PatFac
+
+        def pat_fac_func(shock):
+            return shock ** (1.0 - self.CRRA)
+
+        self.PatFac = (
+            calc_expectation(self.RiskyDstn, pat_fac_func) * self.DiscFacEff
+        ) ** (1.0 / self.CRRA)
+
+        self.MPCminNow = 1.0 / (1.0 + self.PatFac / solution_next.MPCmin)
+
+        self.hNrmNow = (
+            self.PermGroFac / self.Rfree * (self.Ex_IncNext + solution_next.hNrm)
+        )
```

Comment:
@llorracc still thinking about what the appropriate discounting is for human wealth. 
case 1 - divide by `self.RiskyAvg`
case 2 - calculate expectation of `self.PermGroFac * (self.Ex_IncNext + solution_next.hNrm) / risky_shock`

Reply:
I think we decided to impose explicit liquidity constraints, so if I understand things the question is really just about the right interest factor to use to discount to end-of-$t$ the minimum possible income in $t+1$.

If we are going to have only one return factor, the answer is that the minimum notional $a_{t}$ must be such that $a_{t} \times \max{RiskyNrm_{t+1}} = -\min{\theta_{t+1}}$ so we can guarantee with certainty that $a_{t} \times \RiskyNrm_{t+1} + \min{\theta_{t+1}} \geq 0.$

where $\RiskyNrm_{t+1} = \Risky_{t+1}/(\PermGroFac_{t+1} \times \PermShk_{t+1})$ and $\theta$ is the transitory shock.

Reply:
I understand this point, but my question is not related to the borrowing constraint. Human Wealth and the MPCs are used when constructing the limit of the consumption function as in https://github.com/econ-ark/HARK/blob/4bc59842cda28aef6f3382cd6ebca7aa7a5c7df3/HARK/ConsumptionSaving/ConsIndShockModel.py#L731-L732